tools/sqf_validator.py

```diff
@@ -79,39 +124,50 @@ def popClosing():
                         isInCommentBlock = False
                     elif (c != '*'):
                         checkIfNextIsClosingBlock = False
+            indexOfCharacter += 1
 
         if brackets_list.count('[') != brackets_list.count(']'):
-            print("A possible missing square bracket [ or ] in file {0} [ = {1} ] = {2}".format(filepath,brackets_list.count('['),brackets_list.count(']')))
+            print("ERROR: A possible missing square bracket [ or ] in file {0} [ = {1} ] = {2}".format(filepath,brackets_list.count('['),brackets_list.count(']')))
             bad_count_file += 1
         if brackets_list.count('(') != brackets_list.count(')'):
-            print("A possible missing round bracket ( or ) in file {0} ( = {1} ) = {2}".format(filepath,brackets_list.count('('),brackets_list.count(')')))
+            print("ERROR: A possible missing round bracket ( or ) in file {0} ( = {1} ) = {2}".format(filepath,brackets_list.count('('),brackets_list.count(')')))
             bad_count_file += 1
         if brackets_list.count('{') != brackets_list.count('}'):
-            print("A possible missing curly brace {{ or }} in file {0} {{ = {1} }} = {2}".format(filepath,brackets_list.count('{'),brackets_list.count('}')))
+            print("ERROR: A possible missing curly brace {{ or }} in file {0} {{ = {1} }} = {2}".format(filepath,brackets_list.count('{'),brackets_list.count('}')))
             bad_count_file += 1
     return bad_count_file
 
 def main():
 
     print("#########################")
-    print("# Validate SQF files missing brackets  #")
+    print("# Validate SQF #")
     print("#########################")
 
     sqf_list = []
     bad_count = 0
 
     parser = argparse.ArgumentParser()
-    parser.add_argument('-m','--module', help='only search specified module addon folder', required=False, default=".")
+    parser.add_argument('-m','--module', help='only search specified module addon folder', required=False, default="")
     args = parser.parse_args()
 
-    for root, dirnames, filenames in os.walk('../addons' + '/' + args.module):
+    # Allow running from root directory as well as from inside the tools directory
+    rootDir = "../addons"
+    if (os.path.exists("addons")):
+        rootDir = "addons"
+
+    for root, dirnames, filenames in os.walk(rootDir + '/' + args.module):
       for filename in fnmatch.filter(filenames, '*.sqf'):
         sqf_list.append(os.path.join(root, filename))
 
     for filename in sqf_list:
         bad_count = bad_count + check_sqf_syntax(filename)
 
-    print("Bad Count {0}".format(bad_count))
+    print("Checked {0} files. Errors detected {1}.".format(len(sqf_list), bad_count))
```

Comment:
I'd split that into lines for readability and add a colon after errors detected, like so:
```
print("Number of files checked: {0}\nErrors detected: {1}".format(len(sqf_list), bad_count))
```